promql/functions.go

```diff
@@ -149,7 +149,7 @@ func extrapolatedRate(vals []parser.Value, args parser.Expressions, enh *EvalNod
 	if durationToStart >= extrapolationThreshold {
 		durationToStart = averageDurationBetweenSamples / 2
 	}
-	if isCounter && resultFloat > 0 && len(samples.Floats) > 0 && samples.Floats[0].F >= 0 {
+	if isCounter && resultFloat > 0 && len(samples.Floats) > 0 && samples.Floats[0].F > 0 {
```

Comment:
So if `samples.Floats[0].F` is precisely zero, the `durationToZero` below in line 160 is calculated to be zero, so we ultimately set `durationToStart` to zero, or in other word remove any extrapolation to the left.

With your change, we would actually doing the _full_ extrapolation in that case, going far below zero.